### ISSUE TITLE
Open to support alternative implementations of Microsoft Graph?

### ISSUE THREAD

**Issue by fbartels** [PTST/O365Calendar-HomeAssistant#98]
Hi,

I work for a company that is producing an [open source email server](https://kopano.com/products/groupware/integrations-apis/) (with calendar, contacts, tasks) and a while ago we started work on a restapi for our next generation of clients and make it easier for third parties to write integrations against our data. And in an effort to make it even easier for third parties we decided to model our Api based on Microsoft Graph (the api that is driving Office 365).

Long story short, since I am a home assistant users for a while I was wondering if I could connect this to said api and came across your project. Would you be open to accept a pull request that adds the possibility to override the base our for api requests plus the possibility to modify authentication scopes to allow login?
